Clarify template light options (#43936)
home-assistant/home-assistant.github.io: source/_integrations/template.markdown

      type: template
      default: optimistic
    set_effect:

>       description: Defines a set of actions (script) to run when the light is given an effect command. The script will only execute if the light is turned on with an `effect`. The `set_effect` script receives the variable `effect`. It may also receive variables `brightness` and/or `transition`.

Reviewer: Hi! Here are some suggestions:

- Prefer using present tense
- Move "only" just before the condition
- Change "may" to "can" to describe possibility not permission. https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/c/can-may
- Change "and/or" to or because it's clear from the sentence that it's inclusive. https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/a/and-or

```suggestion
      description: Defines a set of actions (script) to run when the light is given an effect command. The script executes only if the light is turned on with an `effect`. The `set_effect` script receives the variable `effect`. It can also receive variables `brightness` or `transition`.
```

      required: inclusive
      type: action
    set_level:

>       description: Defines a set of actions (script) to run when the light is given a brightness command. The script will only execute if the light is turned on with a `brightness`, `brightness_pct`, or `transition`. The `set_level` script receives the variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines a set of actions (script) to run when the light is given a brightness command. The script executes only if the light is turned on with a `brightness`, `brightness_pct`, or `transition`. The `set_level` script receives the variables `brightness` or `transition`.
```

      required: false
      type: action
    set_temperature:

>       description: Defines a set of actions (script) to run when the light is given a color temperature command. The script will only execute if the light is turned on with a `color_temp` or `color_temp_kelvin`. The script receives the variables `color_temp` and `color_temp_kelvin`, and may also receive variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines a set of actions (script) to run when the light is given a color temperature command. The script executes only if the light is turned on with a `color_temp` or `color_temp_kelvin`. The script receives the variables `color_temp` and `color_temp_kelvin`, and can also receive variables `brightness` or `transition`.
```

      required: false
      type: action
    set_hs:

>       description: Defines a set of actions (script) to run when the light is given a hs color command. The script will only execute if the light is turned on with a `hs_color`.  The script receives the variables `hs` as a tuple, `h`, and `s`, and may also receive variables `brightness` and/or `transition`.

Reviewer 2: ```suggestion
      description: Defines a set of actions (script) to run when the light is given a hs color command. The script will only execute if the light is turned on with an `hs_color`.  The script receives the variables `hs` as a tuple, `h`, and `s`, and may also receive variables `brightness` and/or `transition`.
```

>       description: Defines a set of actions (script) to run when the light is given a hs color command. The script will only execute if the light is turned on with a `hs_color`.  The script receives the variables `hs` as a tuple, `h`, and `s`, and may also receive variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines a set of actions (script) to run when the light is given a hs color command. The script executes only if the light is turned on with a `hs_color`. The script receives the variables `hs` as a tuple, `h` and `s`, and can also receive variables `brightness` or `transition`.
```

      required: false
      type: action
    set_rgb:

>       description: Defines a set of actions (script) to run when the light is given an RGB color command. The script will only execute if the light is turned on with a `rgbw_color`.  The script receives the variables `rgb` as a tuple, `r`, `g`, and `b`, and may also receive variables `brightness` and/or `transition`.

Reviewer 2: ```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGB color command. The script will only execute if the light is turned on with an `rgbw_color`.  The script receives the variables `rgb` (a tuple of `r`, `g`, and `b`) and may also receive `brightness` and/or `transition`.
```

>       description: Defines a set of actions (script) to run when the light is given an RGB color command. The script will only execute if the light is turned on with a `rgbw_color`.  The script receives the variables `rgb` as a tuple, `r`, `g`, and `b`, and may also receive variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGB color command. The script executes only if the light is turned on with a `rgbw_color`. The script receives the variables `rgb` as a tuple, `r`, `g`, and `b`, and can also receive variables `brightness` or `transition`.
```

Optionally (but I'm not a dev, so this might be incorrect), use `3-tuple` instead of `tuple`:

```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGB color command. The script executes only if the light is turned on with a `rgbw_color`. The script receives the variables `rgb` as a 3-tuple, `r`, `g`, and `b`, and can also receive variables `brightness` or `transition`.
```

      required: false
      type: action
    set_rgbw:

>       description: Defines a set of actions (script) to run when the light is given an RGBW color command. The script will only execute if the light is turned on with a `rgbw_color`.  The script receives the variables `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, and `w`, and may also receive variables `brightness` and/or `transition`.

Reviewer 2: ```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGBW color command. The script will only execute if the light is turned on with `rgbw_color`.  The script receives the variables `rgbw` and `rgb` as tuples, `r`, `g`, `b`, and `w`, and may also receive `brightness` and/or `transition`.
```

>       description: Defines a set of actions (script) to run when the light is given an RGBW color command. The script will only execute if the light is turned on with a `rgbw_color`.  The script receives the variables `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, and `w`, and may also receive variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGBW color command. The script executes only if the light is turned on with a `rgbw_color`. The script receives the variables `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, and `w`, and can also receive variables `brightness` or `transition`.
```

Or, with clarifying the tuple:

```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGBW color command. The script executes only if the light is turned on with a `rgbw_color`. The script receives the variables `rgbw` as a tuple, `rgb` as a 3-tuple, `r`, `g`, `b`, and `w`, and can also receive variables `brightness` or `transition`.
```

      required: false
      type: action
    set_rgbww:

>       description: Defines a set of actions (script) to run when the light is given an RGBWW color command. The script will only execute if the light is turned on with a `rgbww_color`.  The script receives the variables `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, `cw`, and `ww`, and may also receive variables `brightness` and/or `transition`.

Reviewer 2: ```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGBWW color command. The script will only execute if the light is turned on with a `rgbww_color`.  The script receives the variables `rgbww` and `rgb` as tuples, `r`, `g`, `b`, `cw`, and `ww`, and may also receive `brightness` and/or `transition`.
```

>       description: Defines a set of actions (script) to run when the light is given an RGBWW color command. The script will only execute if the light is turned on with a `rgbww_color`.  The script receives the variables `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, `cw`, and `ww`, and may also receive variables `brightness` and/or `transition`.

Reviewer: ```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGBWW color command. The script executes only if the light is turned on with a `rgbww_color`.  The script receives the variables `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, `cw`, and `ww`, and can also receive variables `brightness` or `transition`.
```

Or, with numbered tuples:

```suggestion
      description: Defines a set of actions (script) to run when the light is given an RGBWW color command. The script executes only if the light is turned on with a `rgbww_color`.  The script receives the variables `rgbww` as a 5-tuple, `rgb` as a 3-tuple, `r`, `g`, `b`, `cw`, and `ww`, and can also receive variables `brightness` or `transition`.
```

      type: template
      default: false
    temperature:

>       description: Defines a template to get the color temperature of the light. The template must return the color temperature in mireds. If you are using a `color_temp_kelvin` attribute from another source, convert the value to mireds by dividing 1000000 by the `color_temp_kelvin` result.

Reviewer: Format the number as code for legibility:

```suggestion
      description: Defines a template to get the color temperature of the light. The template must return the color temperature in mireds. If you are using a `color_temp_kelvin` attribute from another source, convert the value to mireds by dividing `1000000` by the `color_temp_kelvin` result.
```
